### PR TITLE
[4][webservice] user patch

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #38277 .

### Summary of Changes
skip group check when webservices

### Testing Instructions

send a PATCH  to the `{{base_url}}/{{base_path}}/users/{{user_id}}`
Contents:
```json
{"email":"testing@somemail.com", "some_custom_field": "0"}
```

### Actual result BEFORE applying this Pull Request
Error : "Save failed with the following error: You can't save a user account without selecting at least one user group."

### Expected result AFTER applying this Pull Request
for the user specified by the {user_id}, the mail and custom field get updated to the values sent 

### Documentation Changes Required
?

### CODE REVIEW

**Comment by N6REJ** [comment]
your testing instructions are unclear, please give step by step

File: administrator/components/com_users/src/Model/UserModel.php [hunk 1]
```diff
@@ -242,7 +242,7 @@ public function save($data)
         }
 
         // Make sure user groups is selected when add/edit an account
-        if (empty($data['groups']) && ((int) $user->id != (int) $my->id || $iAmSuperAdmin)) {
+        if ((empty($data['groups']) && ((int) $user->id != (int) $my->id || $iAmSuperAdmin)) && (!Factory::getApplication()->isClient('api'))) {
```
Comment (Quy) [inline administrator/components/com_users/src/Model/UserModel.php:245]:
```suggestion
        if (empty($data['groups']) && ((int) $user->id != (int) $my->id || $iAmSuperAdmin)) && !Factory::getApplication()->isClient('api')) {
```

Comment (Quy) [inline administrator/components/com_users/src/Model/UserModel.php]:
```suggestion
        if (empty($data['groups']) && ((int) $user->id != (int) $my->id || $iAmSuperAdmin) && !Factory::getApplication()->isClient('api')) {
```

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on 05d292580be4b21a29b72266123a1b816bb4b21c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38283">issues.joomla.org/tracker/joomla-cms/38283</a>.</sub>

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 4.4-dev.

**Comment by tomsrocket** [comment]
Tried to test, but did'nt get the result: 

My Test: 
curl -d '{"email":"testing@somemail.com", "some_custom_field": "0"}' -X PATCH http://localhost:8080/api/users/51 -H "Authorization: Bearer ..."

Response: 
{"errors":[{"title":"Resource not found","code":404}]}

**Comment by heelc29** [comment]
@tomsrocket It looks like you don't use the correct url. 
In your case it should:
http://localhost:8080/api/index.php/v1/users/51

**Comment by alikon** [comment]
no interest

**Pull request closed without merging**